tests/components/homekit_controller/test_binary_sensor.py

```diff
@@ -107,3 +109,26 @@ def create_occupancy_sensor_service(accessory):
     assert state.state == "on"
 
     assert state.attributes["device_class"] == DEVICE_CLASS_OCCUPANCY
+
+
+def create_leak_sensor_service(accessory):
+    """Define leak characteristics."""
+    service = accessory.add_service(ServicesTypes.LEAK_SENSOR)
+
+    cur_state = service.add_char(CharacteristicsTypes.LEAK_DETECTED)
+    cur_state.value = 0
+
+
+async def test_leak_sensor_read_state(hass, utcnow):
+    """Test that we can read the state of a HomeKit leak sensor accessory."""
+    helper = await setup_test_component(hass, create_leak_sensor_service)
+
+    helper.characteristics[LEAK_DETECTED].value = 0
+    state = await helper.poll_and_get_state()
+    assert state.state == "off"
+
+    helper.characteristics[LEAK_DETECTED].value = 1
+    state = await helper.poll_and_get_state()
+    assert state.state == "on"
+
+    assert state.attributes["device_class"] == DEVICE_CLASS_MOISTURE
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5239303)